src/credentials/sso/diskCache.ts

```diff
@@ -0,0 +1,83 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as fs from 'fs'
+import * as crypto from 'crypto'
+import { SsoCache } from './ssoCache'
+import { ClientRegistration } from './clientRegistration'
+import { join } from 'path'
+import { homedir } from 'os'
+import { AccessToken } from './accessToken'
+
+export class DiskCache implements SsoCache {
+    private cacheDir: string = join(homedir(), '.aws', 'sso', 'cache')
+
+    loadClientRegistration(ssoRegion: string): ClientRegistration | null {
+        if (!this.registrationExists(ssoRegion)) {
+            return null
+        }
+        const registration = JSON.parse(
+            fs.readFileSync(join(this.cacheDir, `aws-toolkit-vscode-client-id-${ssoRegion}.json`)).toString()
+        )
+        if (registration && this.isNotExpired(registration)) {
+            return registration
+        } else {
+            return null
```

Comment:
You might want to treat any caught errors the same as "no registration available". This will gracefully handle scenarios like access restrictions to that file/folder. (Consider similar error handling for save*)